### PR TITLE
MinMax Index Supports Nullable DataType

### LINKED ISSUE #4787: MinMax Index Supports Nullable DataType.
## Enhancement
Currently, MinMax Index does not support `Nullable` datatype. We can extract the nested data type from `DataTypeNullable` and reuse existing MinMax index code.

### PR DESCRIPTION
**Pull request by hehechen**
Signed-off-by: hehechen <awd123456sss@gmail.com>

### What problem does this PR solve?
Currently, MinMax Index does not support Nullable datatype.
Issue Number: close #4787

Problem Summary:

### What is changed and how it works?
We can extract the nested data type from DataTypeNullable and reuse existing MinMax index code.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp [hunk 1]
```diff
@@ -158,6 +155,64 @@ std::pair<UInt64, UInt64> MinMaxIndex::getUInt64MinMax(size_t pack_index)
...
 }
 
+RSResult MinMaxIndex::checkNullableEqual(size_t pack_index, const Field & value, const DataTypePtr & type)
+{
+    const auto & column_nullable = static_cast<const ColumnNullable &>(*minmaxes);
+
+    const auto * raw_type = type.get();
+
+    // if minmaxes_data has null value, the value of minmaxes_data[i] is meaningless and maybe just some random value.
+    // But in checkEqual, we have checked the has_null_marks and ensured that there is no null value in MinMax Indexes.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp:165]: Move these comments to the front of the following lines is more clear
https://github.com/pingcap/tiflash/blob/97cef6476a13e335d0620bf77056a6ab5255548b/dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp#L224-L227

File: dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp [hunk 2]
```diff
@@ -82,14 +80,13 @@ void MinMaxIndex::addPack(const IColumn & column, const ColumnVector<UInt8> * de
         }
     }
 
-    const IColumn & updated_column = *column_ptr;
-    auto [min_index, max_index] = details::minmax(updated_column, del_mark, 0, updated_column.size());
+    auto [min_index, max_index] = details::minmax(column, del_mark, 0, column.size());
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp:83]: What's the `min_index` of we call `details::minmax` with a column `[NULL, 1, 2, NULL]`? Is `min_index` point to the `NULL` or point to the `1`?

Comment (hehechen) [reply dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp:83]: https://github.com/pingcap/tiflash/blob/164eda5296fe0714ba3e7a30f0ffd7a562b378be/dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp#L51 The `null_direction_hint` is `-1`, it means that NULLs are considered as least than everything other, so the `min_index` will point to the `NULL`.

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by jiaqizho — APPROVED** [review]

**Review by lidezhu — APPROVED** [review]
LGTM

**Pull request merged**